### PR TITLE
PLT-1475 Update search query to ignore system messages

### PR DESCRIPTION
**Pull request by jwilander**

### CODE REVIEW

File: store/sql_post_store.go [hunk 1]
```diff
@@ -642,6 +642,7 @@ func (s SqlPostStore) Search(teamId string, userId string, params *model.SearchP
 				Posts
 			WHERE
 				DeleteAt = 0
+				AND Type NOT LIKE '` + model.POST_SYSTEM_MESSAGE_PREFIX + `%'
```
Comment (coreyhulen) [inline store/sql_post_store.go]: does this need to be a like clause?  I think it's going to be very slow.  We might want to load it in a query analyzer and see.

File: api/post_test.go [hunk 2]
```diff
@@ -615,7 +613,7 @@ func TestSearchPostsFromUser(t *testing.T) {
 		t.Fatalf("wrong number of posts returned %v", len(result.Order))
 	}
 
-	if result := Client.Must(Client.SearchPosts("from: " + user2.Username + " in:" + channel1.Name)).Data.(*model.PostList); len(result.Order) != 1 {
+	if result := Client.Must(Client.SearchPosts("from: " + user2.Username + " in:" + channel1.Name)).Data.(*model.PostList); len(result.Order) != 0 {
```
Comment (hmhealey) [inline api/post_test.go]: Can you manually add some posts in place of the system messages for this test instead of knocking the expected numbers down? Some of the tests don't check anything right now like this one which was supposed to check if from: and to: can be used together.

File: api/post_test.go [hunk 3]
```diff
@@ -630,19 +628,19 @@ func TestSearchPostsFromUser(t *testing.T) {
 	// wait for the join/leave messages to be created for user3 since they're done asynchronously
 	time.Sleep(100 * time.Millisecond)
```
Comment (hmhealey) [inline api/post_test.go]: You can take out this sleep as well since we don't need to wait for the join/leave messages

Comment (jwilander) [inline api/post_test.go]: We do want to wait for the join/leave messages because we want to make sure the search doesn't return them

File: store/sql_post_store.go (hunk 1 above)
Comment (jwilander) [inline store/sql_post_store.go]: I can run some analysis on it but I don't think it should be that slow since the field it's checking against isn't very long. Instead of using a LIKE clause I can do post-processing on the rows returned by the query as well, though I'm not sure if that would be faster or slower.

I'll let you know what I find

File: api/post_test.go (hunk 2 above)
Comment (jwilander) [inline api/post_test.go]: Ehh there's only two tests that check for zero counts after my changes and nothing they test is not being tested elsewhere. They're also now checking the negative case where nothing should be returned. If you're not 0/5 I'll make the changes but I don't think it's going to give more coverage

File: api/post_test.go (hunk 3 above)
Comment (hmhealey) [inline api/post_test.go]: Ah good point

File: api/post_test.go (hunk 2 above)
Comment (hmhealey) [inline api/post_test.go]: It's just those two cases that return 0 that concern me since they're the only ones that test using both search flags at once. The other ones are fine as is, but I thought it might be easier to just revert the changes to the expected numbers and add some extra posts programmatically.

File: store/sql_post_store.go (hunk 1 above)
Comment (coreyhulen) [inline store/sql_post_store.go]: It's not the length, it's the wildcard LIKE that's going to cause a table scan (I think).  I think if you look at the query profiler it may show a table scan.  Post processing the returned results will be slower than the LIKE clause in the db.  Plus you could in theory filter the results to zero even though there are valid results because we use LIMIT.

It's probably not a huge deal since I think it will run with the full text index first then scan those vs a full table scan (I think).

For some reason I thought system messages had an ID of `0`, but I forgot we used the type field.  Since ID has an index as well it would have been more performant.

+1

**Comment by crspeller** [comment]
@jwilander Are you still running analysis?

**Comment by jwilander** [comment]
Nope, unless someone feels strongly about it then I wasn't going to

**Comment by crspeller** [comment]
+1
